server/routes/testutils/appSetup.ts

```diff
@@ -122,16 +112,7 @@ function appSetup(
     res.locals = {
       user: { ...req.user },
     }
-    req.session = {
-      ...sessionData,
-      regenerate: jest.fn(),
-      destroy: jest.fn(),
-      reload: jest.fn(),
-      id: 'sessionId',
-      resetMaxAge: jest.fn(),
-      save: jest.fn(),
-      touch: jest.fn(),
-    }
+    req.session = sessionData as Session & Partial<SessionData>
```

Comment:
Improving how `sessionData` is mocked for supertest tests: preserve the object passed in rather than using a new one. Makes it possible to have test assertions for what has changed in the data.